src/ui/components/results/appliedfilterscomponent.js

```diff
@@ -27,7 +28,13 @@ export default class AppliedFiltersComponent extends Component {
     this._verticalKey = this._config.verticalKey ||
       this.core.globalStorage.getState(StorageKeys.SEARCH_CONFIG).verticalKey;
 
-    this.moduleId = StorageKeys.DYNAMIC_FILTERS;
+    this.moduleId = StorageKeys.FACETS_LOADED;
+
+    this.core.globalStorage.on('update', StorageKeys.VERTICAL_RESULTS, results => {
+      if (results.searchState === SearchStates.SEARCH_COMPLETE) {
```

Comment:
For my own edification, we reach the `SEARCH_COMPLETE` state only after new Facets have been added to the registry?

Reply:
This is to handle cases where we don't have a Facets component on the page. I wrote a small explanation and some possible other options above.

Reply:
but this happens before Facets are updated in the registry